src/main/java/com/juniorstart/juniorstart/service/ProjectService.java

```diff
@@ -0,0 +1,250 @@
+ * @author Grzegorz Szczęsny
+ * @version 1.0
+ * @since 1.0
+ */
+@Service
+@Slf4j
+public class ProjectService {
+
+
+    public ProjectService(UserDao userDao, ProjectRepository projectRepository, TechnologiesRepository technologiesRepository) {
+        this.userDao = userDao;
+        this.projectRepository = projectRepository;
+        this.technologiesRepository = technologiesRepository;
+    }
+    private final UserDao userDao;
+    private final ProjectRepository projectRepository;
+    private final TechnologiesRepository technologiesRepository;
+
+    /** Represents an project service which manipulate project data. 26-10-2020.
+     * @author Grzegorz Szczęsny
+     * @version 1.0
+     * @since 1.0
+     */
+    public PagedResponse<ProjectResponse> getAllProjects(int page, int size) {
+        ValidatePageUtil.validatePageNumberAndSize(page, size);
+
+        Pageable pageable = PageRequest.of(page, size, Sort.Direction.DESC, "createdAt");
+        Page<Project> projects = projectRepository.findAll(pageable);
+
+        if(projects.getNumberOfElements() == 0) {
+            return new PagedResponse<>(Collections.emptyList(), projects.getNumber(), projects.getSize(), projects.getTotalElements(), projects.getTotalPages(), projects.isLast());
+        }
+
+        List<ProjectResponse> projectResponses = projects.map(project -> {
+
+            User creator = userDao.findByPublicId(project.getCreatedBy()).orElseThrow(()-> new ResourceNotFoundException("User", "userId", project.getCreatedBy()));
+
+            return ModelMapper.mapProjectToProjectResponse(project, creator);
+        }).getContent();
+
+        return new PagedResponse<>(projectResponses, projects.getNumber(), projects.getSize(), projects.getTotalElements(), projects.getTotalPages(), projects.isLast());
+        
+    }
+
+    /**
+     * Create a new project
+     * @param currentUser A current user's credentials
+     * @param projectRequest A request that contains a credentials for a new project.
+     * @return a new {@link Project}.
+     * @throws BadRequestException if user has too many projects.
+     */
+    public Project createProject(UserPrincipal currentUser, ProjectRequest projectRequest) {
+        User user = userDao.findById(currentUser.getId()).orElseThrow(() -> new ResourceNotFoundException("User", "userId", currentUser.getId()));
+
+        if(userDao.findByEmail(projectRequest.getName()).isPresent()) {
+            throw new BadRequestException("Name already in use.");
+        }
+
+
+            Project newProject = new Project();
+            newProject.setRepository(projectRequest.getRepository());
+            newProject.setName(projectRequest.getName());
+            newProject.setTitle(projectRequest.getTitle());
+            newProject.setNumberOfSeats(projectRequest.getNumberOfSeats());
+            newProject.setDescription(projectRequest.getDescription());
+            newProject.setBody(projectRequest.getBody());
```

Comment:
To bym wyrzucił do jakiejś funkcji

Reply:
Przypisywanie jest potrzebne? Jeśli tak to aby w ten sposób (mam tu na mysli np maper etc)